### ISSUE TITLE
[BUG] Using update, upsert, or delete actions without specifying document_id crashes the pipeline with NPE

### ISSUE THREAD

**Issue by graytaylor0** [opensearch-project/data-prepper#3988]
**Describe the bug**
`update`, `upsert`, and `delete` bulk actions in the opensearch sink requires knowing the `document_id` of the document to update or delete. However, if `document_id` is not provided in the `opensearch` sink with one of these actions being used, the following NPE is hit when converting the failed document to a DLQ object

```
Caused by: java.lang.NullPointerException
	at org.opensearch.dataprepper.plugins.sink.opensearch.dlq.FailedBulkOperationConverter.convertDocumentToGenericMap(FailedBulkOperationConverter.java:64) ~[opensearch-2.6.1.jar:?]
	at org.opensearch.dataprepper.plugins.sink.opensearch.dlq.FailedBulkOperationConverter.convertToDlqObject(FailedBulkOperationConverter.java:38) ~[opensearch-2.6.1.jar:?]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195) ~[?:?]
```

**Expected behavior**
Do not crash the pipeline and send the document to DLQ without any document_id

**Screenshots**

**Environment (please complete the following information):**
 - OS: [e.g. Ubuntu 20.04 LTS]

**Additional context**
Related to #3933

**Comment by dlvenable** [comment]
We should add restrictions to the pipeline configuration to disallow these actions when the customer does not provide a `document_id`.

**Comment by dlvenable** [comment]
Fixed by #4178
